### ISSUE TITLE
Allow for variadic arguments

### ISSUE THREAD

**Issue by pamil** [Behat/Behat#1308]
Since we dropped support for PHP <7.2, all PHP versions we support have variadic arguments implemented. At Sylius we've created https://github.com/FriendsOfBehat/VariadicExtension in the past that allows them to be passed to the step definition. Is it something that we would like to include in the core? Are there any cons?

**Comment by lchrusciel** [comment]
There shouldn't be any problem with transformers, as it just adds all not organized arguments at the end of the function. Basic scenario is covered already: https://github.com/FriendsOfBehat/VariadicExtension/blob/master/features/transforming_variadic_arguments_in_step_definitions.feature

**Comment by ciaranmcnulty** [comment]
Yep I'm interested in scenarios like `(Foo ...$foos)` and the related magic around type-based transformers. If we bring it into core I'd like to cover a bunch more of that with scenarios
